src/main/java/org/openrewrite/java/spring/batch/MigrateJobStep.java

```diff
@@ -0,0 +1,198 @@
+                                .get(0)
+                                .withPrefix(noParams ? Space.EMPTY : Space.SINGLE_SPACE);
+
+                        // Add JobOperator parameter
+                        List<Statement> newParams = noParams ? singletonList(jobOperatorParam) : ListUtils.concat(
+                                filteredParams, jobOperatorParam);
+
+                        List<JavaType> newParamTypes = noParams ? singletonList(jobOperatorParam.getType()) : ListUtils.concat(
+                                md.getMethodType().getParameterTypes(), jobOperatorParam.getType());
+
+                        md = md.withParameters(newParams).withMethodType(md.getMethodType().withParameterTypes(newParamTypes));
+
+                        return super.visitMethodDeclaration(md, ctx);
+                    }
+
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation m, ExecutionContext ctx) {
+                        MethodMatcher methodMatcher = new MethodMatcher(JOB_STEP_JOB_LAUNCHER_SETTER, false);
+                        if (methodMatcher.matches(m)) {
+                            JavaType.Method type = m.getMethodType();
+                            if (type != null) {
+                                type = type.withName("setJobOperator");
+                            }
+                            Expression newArg = renameJobLauncherArgument(m.getArguments().get(0));
+                            m = m.withName(m.getName().withSimpleName("setJobOperator").withType(type)).withMethodType(type)
+                                    .withArguments(singletonList(newArg));
+                        }
+                        return super.visitMethodInvocation(m, ctx);
+                    }
+
+                    private @Nullable Statement cleanConstructor(Statement stmt) {
+                        if (!(stmt instanceof J.MethodDeclaration)) {
+                            return stmt;
+                        }
+
+                        J.MethodDeclaration md = (J.MethodDeclaration) stmt;
+                        if (!md.isConstructor()) {
+                            return stmt;
+                        }
+
+                        // Remove JobLauncher parameter
+                        List<Statement> newParams = md.getParameters().stream()
+                                .filter(p -> !isJobLauncherParameter(p))
+                                .collect(toList());
+
+                        // Remove the assignment: this.jobLauncher = jobLauncher;
+                        J.Block newBody = md.getBody().withStatements(
+                                md.getBody().getStatements().stream()
+                                        .filter(s -> !isJobLauncherAssignment(s))
+                                        .collect(toList())
+                        );
+
+                        J.MethodDeclaration updated = md.withParameters(newParams).withBody(newBody);
```

Comment:
Making a note for myself that we'll probably want to replace these stream usages with `ListUtils` methods `filter` and `map`.